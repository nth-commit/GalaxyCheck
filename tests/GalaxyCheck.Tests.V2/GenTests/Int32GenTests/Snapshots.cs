using System.Collections.Generic;
using System.Linq;
using Xunit;
using GalaxyCheck;
using GalaxyCheck.Gens;
using Snapshooter.Xunit;
using Snapshooter;

namespace Tests.V2.GenTests.Int32GenTests
{
    public class Snapshots
    {
        [Fact]
        public void Values()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IInt32Gen>
            {
                { "Between_0_And_10", Gen.Int32().Between(0, 10) },
                { "Between_0_And_100", Gen.Int32().Between(0, 100) },
                { "Between_0_And_10000", Gen.Int32().Between(0, 10000) },
                { "Between_Default", Gen.Int32() },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var sample = gen.WithBias(bias).Sample(seed: seed);

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }

        [Fact]
        public void ExampleSpaces()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IInt32Gen>
            {
                { "Between_0_And_10", Gen.Int32().Between(0, 10) },
                { "Between_0_And_100", Gen.Int32().Between(0, 100) },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var sample = gen
                    .WithBias(bias)
                    .Advanced.SampleOneExampleSpace(seed: seed, size: 75)
                    .Take(500)
                    .Render(x => x.ToString());

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }
    }
}
