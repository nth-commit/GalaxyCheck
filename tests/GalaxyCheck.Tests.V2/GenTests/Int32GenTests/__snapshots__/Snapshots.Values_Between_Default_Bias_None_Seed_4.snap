[
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  -1,
  -1,
  1,
  0,
  -1,
  -1,
  1,
  4,
  2,
  1,
  2,
  3,
  0,
  -5,
  4,
  1,
  2,
  -5,
  -2,
  0,
  -5,
  10,
  2,
  1,
  -7,
  -9,
  -2,
  -5,
  9,
  8,
  -59,
  -63,
  10,
  -45,
  -32,
  69,
  -19,
  82,
  66,
  -68,
  25,
  -88,
  -2057,
  -685,
  -4527,
  -3993,
  -9115,
  -5578,
  -1287,
  -7903,
  -3358,
  -1926,
  2147483647,
  4285,
  430,
  -91246709,
  -85610267,
  -9604209,
  -4906798,
  -78355170,
  -37481001,
  -13996214,
  -11772516,
  -47119232,
  2147483647,
  -2147483648,
  -34552541,
  2147483647,
  -733227860,
  -439045141,
  -895534523,
  -1723930704,
  -2147483648,
  -1848481938,
  -2124539811,
  -2147483648,
  -1892289910,
  -956836757,
  1848022012,
  553686261,
  -1401266086,
  2147483647,
  1087037735,
  -993031380,
  2021362339,
  755761792,
  2147483647,
  2147483647,
  -2039180220,
  2147483647,
  -1818819247,
  -345905001
]
